pytestqt/wait_signal.py

```diff
@@ -54,11 +57,77 @@ def _quit_loop_by_timeout(self):
             self._loop.quit()
 
     def _cleanup(self):
+        # store timeout message before the data to construct it is lost
+        self._timeout_message = self.get_timeout_error_message()
         if self._timer is not None:
             _silent_disconnect(self._timer.timeout, self._quit_loop_by_timeout)
             self._timer.stop()
             self._timer = None
 
+    def get_timeout_error_message(self):
+        """Subclasses have to implement this, returning an appropriate error message for a SignalTimeoutError."""
+        raise NotImplementedError
+
+    def _extract_pyqt_signal_name(self, potential_pyqt_signal):
```

Comment:
The functions below, which extract a string/name from signal objects, could all live in a `signals.py` module (or something like that) as normal functions instead of methods. It would make it simpler to test each one of them as unit-tests.